apps/prairielearn/src/sync/course-db.ts

```diff
@@ -996,6 +1020,33 @@ function validateQuestion({
     }
   }
 
+  if (question.authors.length > 0) {
+    for (const author of question.authors) {
+      if (author.orcid) {
+        if (!validateORCID(author.orcid)) {
+          errors.push(
+            `The author ORCID identifier ${author.orcid} has an invalid checksum. See the official website (https://orcid.org) for info on how to create or look up an identifier`,
+          );
+        }
+      }
+      if (author.email) {
+        const parsedEmail = z.string().email().safeParse(author.email);
+
+        if (!parsedEmail.success) {
+          errors.push(`The author email address ${author.email} is invalid`);
+        }
+      }
+      if (author.originCourse) {
+        const course = await findCourseBySharingName(author.originCourse);
```

Comment:
This worries me. For a course with 1000s of shared questions with an `originCourse` set, this will explode into thousands of concurrent database queries.

I'd rather limit this to a single query (related: https://github.com/PrairieLearn/PrairieLearn/pull/12816#discussion_r2353207518). This can be done pretty easily as a post-processing step in `loadQuestions` after the `loadInfoForDirectory` call has happened.